src/components/billboard.js

```diff
@@ -0,0 +1,66 @@
+const isMobileVR = AFRAME.utils.device.isMobileVR();
+
+// Billboard component that only updates visible objects and only those in the camera view on mobile VR.
+AFRAME.registerComponent("billboard", {
+  init: function() {
+    this.target = new THREE.Vector3();
+    this._updateBillboard = this._updateBillboard.bind(this);
+    this._updateIsInView = this._updateIsInView.bind(this);
+
+    if (isMobileVR) {
+      this.el.sceneEl.systems["frame-scheduler"].schedule(this._updateIsInView, "billboards");
+    }
+  },
+  remove() {
+    if (isMobileVR) {
+      this.el.sceneEl.systems["frame-scheduler"].unschedule(this._updateIsInView, "billboards");
+    }
+  },
+
+  tick() {
+    if (this.isInView || !isMobileVR) {
+      this._updateBillboard();
+    }
+  },
+
+  _updateIsInView: (function() {
+    const frustum = new THREE.Frustum();
+    const frustumMatrix = new THREE.Matrix4();
+    const cameraWorld = new THREE.Vector3();
+    const isInViewOfCamera = (screenCamera, pos) => {
+      frustumMatrix.multiplyMatrices(screenCamera.projectionMatrix, screenCamera.matrixWorldInverse);
+      frustum.setFromMatrix(frustumMatrix);
+      return frustum.containsPoint(pos);
```

Comment:
This seems kind of suspicious -- doesn't it mean that the thing will be considered in view only iff the center is in view? It seems better to check whether the bounding box or sphere of the entity intersects the frustum.

Reply:
that's true. this code is lifted from the ik controller check, i don't remember if i had a reason to avoid the box check. perhaps concerns about performance or complexity. in this case, the visual artifact is more profound than in the avatar case so i think a box check is probably warranted.

Reply:
updated to do this. not worried about performance here because the box check happens in the scheduled frame.